mbzirc_ign/launch/competition.launch.py

```diff
@@ -0,0 +1,171 @@
+
+    for model in models:
+        ignition_spawn_entity = Node(
+            package='ros_ign_gazebo',
+            executable='create',
+            output='screen',
+            arguments=model.spawn_args()
+        )
+        launch_processes.append(ignition_spawn_entity)
+
+        bridges, nodes = model.bridges(world_name)
+
+        if model.isUAV():
+            payload = model.payload_bridges(world_name)
+            payload_bridges = payload[0]
+            payload_nodes = payload[1]
+            payload_launches = payload[2]
+            bridges.extend(payload_bridges)
+            nodes.extend(payload_nodes)
+            launch_processes.extend(payload_launches)
+
+        if model.isFixedWingUAV():
+            nodes.append(Node(
+                package='mbzirc_ros',
+                executable='fixed_wing_bridge',
+                output='screen',
+                parameters=[{'model_name': model.model_name}],
+            ))
+
+        nodes.append(Node(
+            package='ros_ign_bridge',
+            executable='parameter_bridge',
+            output='screen',
+            arguments=[bridge.argument() for bridge in bridges],
+            remappings=[bridge.remapping() for bridge in bridges],
+        ))
+
+        # tf broadcaster
+        nodes.append(Node(
+            package='mbzirc_ros',
+            executable='pose_tf_broadcaster',
+            output='screen',
+            parameters=[
+                {'world_frame': world_name}
+            ]
+        ))
+
+        # video target relay
+        nodes.append(Node(
+            package='mbzirc_ros',
+            executable='video_target_relay',
+            output='screen',
+            parameters=[{'model_name': model.model_name}]
+        ))
+
+        group_action = GroupAction([
+            PushRosNamespace(model.model_name),
+            *nodes
+        ])
+
+        handler = RegisterEventHandler(
+            event_handler=OnProcessExit(
+                target_action=ignition_spawn_entity,
+                on_exit=[group_action],
+            )
+        )
+        launch_processes.append(handler)
+    return launch_processes
+
+
+def launch_competition_bridges():
+    bridges = [
+        mbzirc_ign.bridges.score(),
+        mbzirc_ign.bridges.clock(),
+        mbzirc_ign.bridges.run_clock(),
+        mbzirc_ign.bridges.phase(),
+        mbzirc_ign.bridges.stream_status(),
+    ]
+    nodes = []
+    nodes.append(Node(
+        package='ros_ign_bridge',
+        executable='parameter_bridge',
+        output='screen',
+        arguments=[bridge.argument() for bridge in bridges],
+        remappings=[bridge.remapping() for bridge in bridges],
+    ))
+    return nodes
+
+
+def launch_vehicles(context, *args, **kwargs):
+    config_file = LaunchConfiguration('config_file').perform(context)
+    world_name = LaunchConfiguration('world').perform(context)
+    return spawn(context, config_file, world_name)
```

Comment:
is it possible to reuse `spawn_config.launch.py` here instead of another spawn function?